gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java

```diff
@@ -118,7 +117,19 @@ public Blob(Storage storage, BlobInfo info) {
    */
   public Blob(Storage storage, String bucket, String blob) {
     this.storage = checkNotNull(storage);
-    this.info = BlobInfo.of(checkNotNull(bucket), checkNotNull(blob));
+    this.info = BlobInfo.builder(BlobId.of(bucket, blob)).build();
```

Comment:
I am not sure either. I have a slight preference for keeping it as is and let the user decide when to fetch data.
If we decide to do get from the service I suggest we do it in any case, not only when `BlobInfo` is not provided (consider the case when `BlobInfo.builder("b", "n").build()` is used).